### PR TITLE
Move IoT az_result and log classification enum values to separate headers without introducing new typedefs

### PR DESCRIPTION
**Pull request by ahsonkhan**
Fixes https://github.com/Azure/azure-sdk-for-c/issues/1190

Follow up from https://github.com/Azure/azure-sdk-for-c/pull/1275 as an alternative approach, which is hopefully less disruptive, but still achieves the desired goal.

Using this approach over the alternative to hit a balance between API design and documentation/IntelliSense experience: https://github.com/Azure/azure-sdk-for-c/pull/1295

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]

**Review by ericwol-msft — APPROVED** [review]

File: sdk/inc/azure/core/az_result.h [hunk 1]
```diff
@@ -38,7 +38,8 @@ enum
 #define _az_RESULT_MAKE_ERROR(facility, code) \
   ((int32_t)((uint32_t)_az_ERROR_FLAG | ((uint32_t)(facility) << 16U) | (uint32_t)(code)))
```
Comment (antkmsft) [inline sdk/inc/azure/core/az_result.h:39]: let's cast this to az_result itself?

File: sdk/inc/azure/core/az_result.h [hunk 2]
```diff
@@ -38,7 +38,8 @@ enum
...
 
-#define _az_RESULT_MAKE_SUCCESS(facility, code) (((uint32_t)(facility) << 16U) | (uint32_t)(code))
+#define _az_RESULT_MAKE_SUCCESS(facility, code) \
+  ((int32_t)(((uint32_t)(facility) << 16U) | (uint32_t)(code)))
```
Comment (antkmsft) [inline sdk/inc/azure/core/az_result.h:42]: let's cast this to az_result itself?

**Review by antkmsft — APPROVED** [review]
👍

**Comment by ahsonkhan** [comment]
@JeffreyRichter please take a look

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**